test/sanity/validate-modules/utils.py

```diff
@@ -57,8 +56,6 @@ def __enter__(self):
         self.sys_stderr = sys.stderr
         sys.stdout = self.stdout = StringIO()
         sys.stderr = self.stderr = StringIO()
-        setattr(sys.stdout, 'encoding', self.sys_stdout.encoding)
-        setattr(sys.stderr, 'encoding', self.sys_stderr.encoding)
```

Comment:
These lines were added on purpose, and will cause failure in Ansible itself.  The line above about not using cStringIO due to needing this setattr also eludes to this.

I didn't explicitly outline it in the original commit where it was added, but not having `encoding` on `sys.stdout` and `sys.stderr` caused a traceback in ansible itself when calling `get_docstring` when calling `Display.display`.

I'm not sure if python3 will require the fileobj to have an `encoding` attribute, but if it does, we'll probably have to look at using StringIO for py2, and BytesIO for py3.